dll/shellext/ntobjshex/foldercommon.h

```diff
@@ -452,16 +452,15 @@ class CCommonFolder :
 
             if (cidl == 1 && IsFolder(apidl[0]))
             {
-                res = RegOpenKey(HKEY_CLASSES_ROOT, L"Folder", keys + 0);
+                res = RegOpenKeyEx(HKEY_CLASSES_ROOT, L"Folder", 0, KEY_READ, &keys[0]);
                 if (!NT_SUCCESS(res))
-                    return HRESULT_FROM_NT(res);
-            }
-            else
-            {
-                nkeys = 0;
+                    return HRESULT_FROM_WIN32(res);
+                nkeys++;
             }
 
             HRESULT hr = CDefFolderMenu_Create2(parent, hwndOwner, cidl, apidl, psfParent, DefCtxMenuCallback, nkeys, keys, &pcm);
+            for (UINT i = 0; i < nkeys; ++i)
+                RegCloseKey(keys[0]);
```

Comment:
Not `RegCloseKey(keys[i]);`  ?
(Well _**OK**_, after reading 10+ lines of code above, I noticed that the array has only one element, and nKeys is either 0 or 1, but making a whole for-loop just to close one single element is unnatural.... So I would strongly suggest you change this.)